components/collection/drop/GenerativePreview.vue

```diff
@@ -180,15 +180,11 @@ watch(imageDataLoaded, () => {
   }
 })
 
-watch(
-  accountId,
-  () => {
-    generateNft()
-  },
-  {
-    immediate: true,
-  },
-)
+watch(accountId, generateNft)
+
+onMounted(() => {
```

Comment:
do these changes fix the issue? it was reloaded because of this line right? https://github.com/kodadot/nft-gallery/blob/684296e5550708088f24ef1863f8c26b0169ce73/components/collection/drop/GenerativePreview.vue#L192-L194

Reply:
>do these changes fix the issue?

yes, that's why i made them

please, try for yourself 

>it was reloaded because of this line right

yes, but simple removal of it introduces the problem you solved by adding it

addition of the timeout causes your watcher to never trigger as far as i could tell while playing with it

if you wanna YOLO we can remove the watcher

Reply:
I think it was fixed with your changes on `components/collection/drop/GenerativeLayout.vue`, right? so, no need this changes?

Reply:
> addition of the timeout causes your watcher to never trigger as far as i could tell while playing with it

I couldn't see any difference in this. It seems like both still trigger the watcher. Or is there a code example I can test?

Reply:
1. without it, first image shown is always same (not a unique variation)
![image](https://github.com/kodadot/nft-gallery/assets/22791238/82963dc6-e503-489c-b29a-e14ac2408424)

2.  with `immediate: true`, this happens often (not always)
```js
watch(accountId, generateNft, { immediate: true })

// onMounted(() => {
//   setTimeout(generateNft, 1000)
// })
```

![image](https://github.com/kodadot/nft-gallery/assets/22791238/5f97b7b1-f59b-4eab-a783-c136ddfad639)


**3. with setTimeout, first image is unique and loads well**

Reply:
Oh, maybe it's way harder to reproduce? It already looks good on my local if I revert this file. But, let's proceed with this anyway

Reply:
i get why you don't like the `setTimeout`
I'm not too big on it either

but it produces better end result in this case

Reply:
> i get why you don't like the setTimeout
> without it, first image shown is always same (not a unique variation)

nah, I'm good as long it can fix the issue. I think it's because you already fixed the important issue with the race condition on emit, seems like it's quite harder to reproduce (good thing). It looks good on my local, even though I always get a unique variant if I revert this file. 🤣

> but it produces better end result in this case

yes, I don't want to take a risk. Let's continue with it.